space_invaders.py

```diff
@@ -220,20 +237,22 @@ def update_alien_positions(screen, alien_fleet, alien_group, counter):
         The total number of ticks elapsed since the game started.
 
     """
-    if alien_velocity[0] < 0: # fleet moving left
+    if Alien.velocity[0] < 0: # fleet moving left
         for alien_row in alien_fleet: # move and redraw aliens
             for alien in alien_row:
                 if (counter % ALIEN_MOVE_PERIOD) == 0:
+                    Alien.toggle_image()
                     alien.update()
     else: # fleet moving right
         for alien_row in alien_fleet:
             for alien in alien_row[::-1]: # reverse row so that the first
                 # alien to contact the window boundaries will be updated
                 # first
                 if (counter % ALIEN_MOVE_PERIOD) == 0:
+                    Alien.toggle_image()
                     alien.update()
-    if alien_velocity[1]:
-        alien_velocity[1] = 0
+    if Alien.moving_down:
```

Comment:
I dont think using static variables in Alien is the right way to enforce all aliens following the same direction/velocity. IE: consider two different fleets of aliens. One going vertically and one going vertically. An AlientFleet class probably makes more sense, and then each alien would belong to an alien fleet. But maybe this is overengineering, idk